components/script/dom/webidls/MediaDevices.webidl

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+// https://w3c.github.io/mediacapture-main/#dom-mediadevices
+
+[Exposed=Window,
+SecureContext, Pref="dom.webrtc.enabled"]
+interface MediaDevices : EventTarget {
+    //                attribute EventHandler ondevicechange;
+    // Promise<sequence<MediaDeviceInfo>> enumerateDevices();
+};
+
+partial interface Navigator {
+    // [SameObject, SecureContext]
+    readonly        attribute MediaDevices mediaDevices;
+};
+
+partial interface MediaDevices {
+    // MediaTrackSupportedConstraints getSupportedConstraints();
+    Promise<MediaStream> getUserMedia(optional MediaStreamConstraints constraints);
```

Comment:
Pref.

Reply:
no need, MediaDevices is pref-gated already